### ISSUE TITLE
Using relational operators (>) with event-properties isn't working correctly

### ISSUE THREAD

**Issue by aireq** [NLog/NLog#1727]
Is it it possible to setup a condition with a numeric comparison to an event property?

I want to track 'Activation' events in my application that take over some threshold time to complete.

The following will filter out all 'Activation' events.

```
<when condition="'${event-properties:item=Event}' == 'Activation'" action="Ignore"/>
```

Adding `and 100  > 10` breaks this filter as expected, and allows all the messages through.

```
<when condition="'${event-properties:item=Event}' == 'Activation' and  100 > 10" action="Ignore"/>
```

Changing this to `100 > 1000` and once again all 'Activation' events are filtered out.

```
<when condition="'${event-properties:item=Event}' == 'Activation' and  100 > 1000 " action="Ignore"/>
```

Then I tried adding my `Elapsed` event property.

```
<when condition="'${event-properties:item=Event}' == 'Activation' and  100 >  '${event-properties:item=Elapsed}'" action="Ignore"/>
```

But this still continues to log all Activation events, including those that completed in less then 100 ms.

Is this this an issue with data types? Is is my Elapsed value being compared to 100 as a string rather then a number? If so is there some way to data type conversions in a condition?

**Comment by 304NotModified** [comment]
To summarize, `10< 100 ` works, but `${event-properties:item=Elapsed}` < 100` not, isn't it?

**Comment by 304NotModified** [comment]
Confirm, this is an issue. 

See unit test . https://github.com/NLog/NLog/pull/1730

The problem is the ordering in ConditionRelationalExpression.PromoteTypes [see code](https://github.com/NLog/NLog/blob/258bb8fa67ebd2bb69b236f8f9886a3195bc561c/src/NLog/Conditions/ConditionRelationalExpression.cs#L155-L160).

Because one of them is a string, the other will be "promoted" to string. 

The quick fix is to change the ordering. But this is problematic as `100 < 'a`` will give an error then.

So we should reimplement it,  a more try approach. Also the code could be more efficient also. I will try to do so, but too bad no time more today for it.

**Comment by 304NotModified** [comment]
This has been fixed in 4.3.11: https://www.nuget.org/packages/NLog/4.3.11
